### PR TITLE
Update the rest of Instructor*.js to ES6 #6819

### LINKED ISSUE #6819: Update the rest of Instructor*.js to ES6
Migrate remaining instructor* to ES6: instructorHome, InstructorSearch and InstructorStudent*.js

### PR DESCRIPTION
**Pull request by Neurrone**
Fixes #6819

Outline of solution:

* Update the rest of instructor*.js to ES6, making only minimal changes necessary to pass ESLint without errors

### CODE REVIEW

**Comment by Neurrone** [comment]
@thenaesh ready for review.

**Comment by junhaoyap** [comment]
Ping @thenaesh

File: src/main/webapp/js/instructorStudentRecords.es6 [hunk 1]
```diff
@@ -122,6 +126,8 @@ function checkComment(form) {
     }
 }
 
+/* eslint-enable consistent-return */
```
Comment (wkurniawan07) [inline src/main/webapp/js/instructorStudentRecords.es6]: You don't need to do this. Just replace `return true;` with `return;` and `return false;` with `e.preventDefault()`, ESLint should not complain anymore.

Comment (thenaesh) [inline src/main/webapp/js/instructorStudentRecords.es6]: Warning: Replace `return false;` in event handlers with `e.preventDefault(); e.stopPropagation();`.

I just got burned by this.

Comment (wkurniawan07) [inline src/main/webapp/js/instructorStudentRecords.es6]: I know about the `e.stopPropagation()` thing, but I'm pretty sure most of the time whoever put `return false;` there only intended it to be `e.preventDefault()`.

Of course exceptions happen. Do check out if `e.stopPropagation()` is also needed.

File: src/main/webapp/js/instructorStudentRecords.es6 [hunk 2]
```diff
@@ -102,23 +103,23 @@ function submitCommentForm(commentIdx) {
...
-        formTextField = tinymce.get('commentText' + editorId).getContent();
+        const editorId = $(form).attr('id').match(/^form_commentedit-(\d+)$/)[1];
+        formTextField = tinymce.get(`commentText${editorId}`).getContent();
     }
 
     if (isBlank(formTextField)) {
         setStatusMessage(DISPLAY_COMMENT_BLANK, StatusType.DANGER);
         scrollToTop();
-        return false;
+        form.preventDefault();
```
Comment (wkurniawan07) [inline src/main/webapp/js/instructorStudentRecords.es6]: Wait a minute, `form` is not an event object. In fact, there is no event object involved here. (so this `return false;` should have been doing nothing all these while?)

File: src/main/webapp/js/instructor.es6 [hunk 3]
```diff
@@ -58,76 +59,76 @@ function isStudentInputValid(editName, editTeamName, editEmail) {
...
             },
-            success: function(data) {
+            success(data) {
                 $('#courseList').html(data);
                 // If the user alt-clicks, the form does not send any parameters and results in an error.
                 // Prevent default form submission and submit using jquery.
                 $('#fscopy_submit').off('click')
                                    .on('click',
-                                        function(event) {
+                                        (e) => {
```
Comment (li-kai) [inline src/main/webapp/js/instructor.es6]: Could you align this with the `.on('click',` above?

File: src/main/webapp/js/instructor.es6 [hunk 4]
```diff
@@ -58,76 +59,76 @@ function isStudentInputValid(editName, editTeamName, editEmail) {
...
+            },
         });
     });
 
     $('#instructorCopyModalForm').submit(
-        function(e) {
+        function (e) {
             e.preventDefault();
-            var $this = $(this);
+            const $this = $(this);
```
Comment (li-kai) [inline src/main/webapp/js/instructor.es6]: `const $this = $(e.target);` and we can make this function an arrow func :D

**Comment by Neurrone** [comment]
@li-kai deferring to future PR, as agreed.

**Pull request merged**